server/api/core/import.js

```diff
@@ -410,35 +402,26 @@ Import.object = function (collection, key, object) {
   // hooks for additional import manipulation.
   const importObject = Hooks.Events.run(`onImport${this._name(collection)}`, object);
 
-  // Clone object for cleaning
-  const cleanedObject = Object.assign({}, importObject);
-
   // Cleaning the object adds default values from schema, if value doesn't exist
-  collection.simpleSchema(importObject).clean(cleanedObject);
+  const cleanedModifier = collection.simpleSchema(importObject).clean({
+    $set: importObject
+  }, {
+    isModifier: true,
+    extendAutoValueContext: { isUpsert: true }
+  });
 
   // And validate the object against the schema
-  this.context(collection, updateObject).validate(cleanedObject, {});
-
-  // Disjoint importObject and cleanedObject again
-  // to prevent `Cannot update '<field>' and '<field>' at the same time` errors
-  const defaultValuesObject = doRightJoinNoIntersection(importObject, cleanedObject);
+  this.context(collection, importObject).validate(cleanedModifier, {
+    modifier: true,
+    upsert: true
+  });
 
   // Upsert the object.
-  const find = this.buffer(collection).find(key);
-
   // With the upsert option set to true, if no matching documents exist for the Bulk.find() condition,
   // then the update or the replacement operation performs an insert.
   // https://docs.mongodb.com/manual/reference/method/Bulk.find.upsert/
-  if (Object.keys(defaultValuesObject).length === 0) {
-    find.upsert().update({
-      $set: importObject
-    });
-  } else {
-    find.upsert().update({
-      $set: importObject,
-      $setOnInsert: defaultValuesObject
-    });
-  }
+  this.buffer(collection).find(key).upsert().update(cleanedModifier);
+
```

Comment:
I'm doing all of the above in a different way, but I'm pretty sure it has the same end effect